packages/tsx-editor/src/transpiler/exampleTransform.test.ts

```diff
@@ -21,6 +21,146 @@ describe('example transform', () => {
 
   it('returns an error from relative imports', () => {
     const result = transformFile('relativeImport.txt');
-    expect(result.error).toBe('error: unsupported imports');
+    expect(result.error).toBe("Error while transforming example: Unsupported import - import '../../fake.scss';.");
+  });
+
+  it('handles transpiled examples with function components', () => {
+    const result = transformFile('functionTranspiled.txt');
+    expect(result.output).toMatchSnapshot();
+  });
+
+  it('handles transpiled examples with class components', () => {
+    const result = transformFile('classTranspiled.txt');
+    expect(result.output).toMatchSnapshot();
+  });
+
+  it('returns an error from relative imports in code that was transpiled', () => {
+    const result = transformFile('relativeImportTranspiled.txt');
+    expect(result.error).toBe("Error while transforming example: Unsupported import - import '../../fake.scss';.");
+  });
+
+  it('detects class names', () => {
+    const classNamePatternCopy = new RegExp(classNamePattern.source);
+    // This is needed to get rid of the global flag from the RegExp
+    expect(`var SpinButtonBasicExample = /** @class */ (function (_super) {
+      __extends(SpinButtonBasicExample, _super);
+      function SpinButtonBasicExample() {
+          return _super !== null && _super.apply(this, arguments) || this;
+      }
+      SpinButtonBasicExample.prototype.render = function () {
+          return (React.createElement("div", { style: { width: '400px' } },
```

Comment:
Maybe to make the tests more readable, you can simplify the render functions down to just `return React.createElement("div");`